nixos/modules/services/web-servers/garage.nix

```diff
@@ -6,6 +6,16 @@ let
   cfg = config.services.garage;
   toml = pkgs.formats.toml { };
   configFile = toml.generate "garage.toml" cfg.settings;
+  wrapper = pkgs.writeScriptBin "garage" ''
+    # make it so all future variables set are automatically exported as environment variables
+    set -a
+
+    # source the set environmentFile (since systemd EnvironmentFile is supposed to be a minor subset of posix sh parsing) (with shell arg escaping to avoid quoting issues)
+    . ${lib.escapeShellArg cfg.environmentFile}
```

Comment:
nit: I'd move the comments in the Nix code rather than in the shell script, they serve developers reading the code and trying to understand the intents.

Reply:
I just moved the wrapper inline again since we don't have branching.